multiqc/modules/ribotish/ribotish.py

```diff
@@ -0,0 +1,398 @@
+
+            # Convert to valid Python dictionary format
+            input_string = input_string.replace("{", "").replace("}", "")
+            input_string = re.sub(r"(\d+):\s*\[", r'"\1": [', input_string)
+            input_string = re.sub(r"\s+", " ", input_string)
+            input_string = "{" + input_string.strip() + "}"
+
+            # Parse the dictionary
+            input_string = input_string.replace("'", '"')
+            input_string = re.sub(r"(\d+),\s*(?=\d)", r"\1, ", input_string)
+            ribo_seq_counts = ast.literal_eval(input_string)
+
+            # Convert keys to integers for consistent handling
+            return {int(k): v for k, v in ribo_seq_counts.items()}
+
+        except (SyntaxError, ValueError, IndexError) as e:
+            log.warning(f"Error parsing {f['fn']}: {e}")
+            return {}
+
+    def calculate_frame_proportions(self):
+        """Calculate frame proportions for each sample and read length."""
+        for sample_name, ribo_seq_counts in self.ribotish_data.items():
+            self.frame_proportions[sample_name] = {}
+
+            for length, counts in ribo_seq_counts.items():
+                total = sum(counts)
+                if total > 0:
+                    self.frame_proportions[sample_name][length] = {
+                        "f0_prop": counts[0] / total,
+                        "f1_prop": counts[1] / total,
+                        "f2_prop": counts[2] / total,
+                        "total": total,
+                    }
+                else:
+                    self.frame_proportions[sample_name][length] = {
+                        "f0_prop": 0,
+                        "f1_prop": 0,
+                        "f2_prop": 0,
+                        "total": 0,
+                    }
+
+    def add_frame_proportion_bargraph(self):
+        """
+        Create a stacked bar graph showing frame proportions for all read lengths.
+        All read lengths are shown side-by-side in a single plot for visual comparison.
+        Each sample-length combination gets its own bar.
+        """
+        # First, collect all read lengths across all samples
+        all_lengths = set()
+        for sample_data in self.frame_proportions.values():
+            all_lengths.update(sample_data.keys())
+        all_lengths = sorted(all_lengths)
+
+        # Create a single dataset with all sample-length combinations
+        # Group by read length first, then by sample within each length
+        plot_data = {}
+        for i, length in enumerate(all_lengths):
+            # Add spacer before each group (except first) for visual separation
+            if i > 0:
+                # Use spaces to create unique but minimal spacer keys
+                spacer_key = " " * i
+                plot_data[spacer_key] = {
+                    "Frame 0": 0,
+                    "Frame 1": 0,
+                    "Frame 2": 0,
+                }
```

Comment:
ok so you get the groups by adding in empty samples? Simple, I like it.

It'll make the data export a bit ugly, but that's probably tolerable I guess.